src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,30 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (object == null) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle)object;
+        if (object.getClass().equals(Rectangle.class)) {
+            return width.equals(rectangle.width)
+                    && length.equals(rectangle.length)
+                    && color.equals(rectangle.color);
+        }
+        if (this == object) {
```

Comment:
use this check at the beginning of the method, it will be better for performance